### ISSUE TITLE
Can't exit debug magic after using a completion

### ISSUE THREAD

**Issue by ngoldbaum** [ipython/ipython#9611]
This is under IPython 5.0b3. Here's a sample session that illustrates the issue:

```
In [1]: def raise_error():
   ...:     raise RuntimeError
   ...:
   ...:

In [2]: raise_error()
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-2-b471b5468223> in <module>()
----> 1 raise_error()

<ipython-input-1-dafa64cbfa5c> in raise_error()
      1 def raise_error():
----> 2     raise RuntimeError
      3

RuntimeError:

In [3]: %debug
> <ipython-input-1-dafa64cbfa5c>(2)raise_error()
      1 def raise_error():
----> 2     raise RuntimeError
      3

ipdb> RuntimeError
<type 'exceptions.RuntimeError'>
ipdb> quit()
ipdb> quit
<IPython.core.autocall.ExitAutocall object at 0x10d534c90>
ipdb> quit()
ipdb> quit
<IPython.core.autocall.ExitAutocall object at 0x10d534c90>
ipdb> exit
<IPython.core.autocall.ExitAutocall object at 0x10d534c90>
ipdb> exit
<IPython.core.autocall.ExitAutocall object at 0x10d534c90>
ipdb> exit
<IPython.core.autocall.ExitAutocall object at 0x10d534c90>
ipdb> exit
<IPython.core.autocall.ExitAutocall object at 0x10d534c90>
```
I was eventually able to exit out of IPython completely using ctrl-c.

**Comment by ngoldbaum** [comment]
Here I generated the line that just executes `RuntimeError` by using a completion after typing "Runtime".

**Comment by takluyver** [comment]
The `q` short command still works. I think the completion is a red herring here. The likely culprit is the 'smart command mode' introduced in #9449 (pinging @anntzer). When variable names collide with commands, 'smart command mode' prefers the variable name, and you can force the command with `!!quit`.

Whether variables should beat commands in general is probably a matter of opinion, but I don't think it makes much sense to display our special `ExitAutocall` object which is available as `quit` and `exit`.

**Comment by anntzer** [comment]
Perhaps we can exclude entries that are in `user_ns_hidden`?  Or just special case `quit` away.

**Comment by Carreau** [comment]
> exclude entries that are in user_ns_hidden

That's likely a good thing to do when in debugger,

**Comment by takluyver** [comment]
I'm a bit wary of using `user_ns_hidden` because it's not a very visible feature, and it might be confusing when some inputs work one way and some work another way. Maybe for now we should just special-case quit and exit. We can always do something smarter later.

**Comment by Carreau** [comment]
fixed by #9622
